### DISCUSSION TITLE
Vaulting added in cpp : no smooth root motion

### DISCUSSION THREAD

**Discussion by MrWilsonWilson** [Sixze/ALS-Refactored#429] [General]
hey there, Ive bene adding some trace checks in the c++ and in Low mantle mode Basically if the obstacle is thin, I perform a montage for vaulting similarly to mantling.

Now the montage plays, and the target location is proper, but unlike the mantling, I get the capsule instantly taken to the target location instead of upon completing the montage

https://github.com/Sixze/ALS-Refactored/assets/122884397/84eaf940-7f04-44e2-a379-ca10f70d2d66

Blue sphere debug being the target location

![image](https://github.com/Sixze/ALS-Refactored/assets/122884397/4529ea29-8887-4212-8788-40cebc1cf4b8)
![image](https://github.com/Sixze/ALS-Refactored/assets/122884397/ae171158-f1c0-4531-a51b-9c0580bc2d39)
![image](https://github.com/Sixze/ALS-Refactored/assets/122884397/3df6e738-1dcf-4416-9742-3ea8f1a5ea26)

Which parameter here determines the rate at which we reach destination ?

![image](https://github.com/Sixze/ALS-Refactored/assets/122884397/afd9ffef-8d54-4421-b4a9-666a90a0167f)

How is the smooth interpolation of the capsule to the target location applied in there ? Thank you for your time and amazing work ! your latest update helped a lot on our legs issues, and here is the trailer of the game we are making using that plugin 👍 

https://www.youtube.com/watch?v=PzGxHbono7w

**Comment by colorindarkness** [comment]
I'm also kind of stuck here with the same issue.

**Reply by MrWilsonWilson** [reply]
Might be something to do with a Z target check, because mantling returns a different Z, but vaulting returns a similar Z (depending on how its setup)
So maybe a check doesn't pass there, but I haven't investigated more, busy with other things
